codalab/lib/upload_manager.py

```diff
@@ -431,8 +440,11 @@ def upload_to_bundle_store(
             bundle_read_str = data.get('bundle_read_url', bundle_url)
             try:
                 progress = FileTransferProgress('Sent ', f=self.stderr)
+                upload_func = self.upload_Azure_blob_storage
+                if bundle_url.startswith(StorageURLScheme.GCS_STORAGE.value):
+                    upload_func = self.upload_GCS_blob_storage
```

Comment:
We should store a dictionary that maps `StorageURLScheme.GCS_STORAGE.value` to `self.upload_GCS_blob_storage`, etc. We should also throw an error if the bundle URL matches neither Azure nor GCS.